src/AppInstallerCLICore/COMContext.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "COMContext.h"
+
+namespace AppInstaller
+{
+    void COMContext::BeginProgress()
+    {
+        m_comProgressCallback(ReportType::BeginProgress, m_current, m_maximum, m_type, m_executionPhase);
+    };
+
+    void COMContext::OnProgress(uint64_t current, uint64_t maximum, ProgressType type)
+    {
+        SetProgress(current, maximum, type);
+        m_comProgressCallback(ReportType::Progressing, m_current, m_maximum, m_type, m_executionPhase);

```

Comment:
>m_comProgressCallback [](start = 8, length = 21)

A note on the fun that this will actually be; we will probably want to have it signaling another thread to actually call out to the COM progress call.  Otherwise we are looking at a blocking OOP COM call on every progress report, which will slow things down a lot.  Nothing to do right now, but fun for someone in the future.